clients/src/main/java/org/apache/kafka/clients/producer/KafkaProducer.java

```diff
@@ -960,6 +973,8 @@ private void throwIfProducerClosed() {
                 log.trace("Waking up the sender since topic {} partition {} is either full or getting a new batch", record.topic(), partition);
                 this.sender.wakeup();
             }
+            long endTime = time.milliseconds();
```

Comment:
this is not the end-to-end latency; I think the original ticket meant the latency between send and actual ack; this is just the latency to enqueue to the sender thread.